tests/test_entities.py

```diff
@@ -1,223 +1,60 @@
-
-        """
-        with self.assertRaises(NoSuchPathError):
-            entity(self.server_config).path(which=path)
-
-    def test_foremantask_path(self):
-        """Test :meth:`nailgun.entities.ForemanTask.path`.
-
-        Assert that correct paths are returned when:
-
-        * an entity ID is provided and the ``which`` argument to ``path`` is
-          omitted
-        * ``which = 'bulk_search'``
-
-        """
-        self.assertIn(
-            '/foreman_tasks/api/tasks/{0}'.format(self.id_),
-            entities.ForemanTask(self.server_config, id=self.id_).path()
-        )
-        for gen_path in (
-                entities.ForemanTask(self.server_config).path(
-                    which='bulk_search'
-                ),
-                entities.ForemanTask(self.server_config, id=self.id_).path(
-                    which='bulk_search'
-                )
-        ):
-            self.assertIn('/foreman_tasks/api/tasks/bulk_search', gen_path)
-
-    def test_syncplan_path(self):
-        """Test :meth:`nailgun.entities.SyncPlan.path`.
-
-        Assert that the correct paths are returned when the following paths are
-        provided to :meth:`nailgun.entities.SyncPlan.path`:
-
-        * ``add_products``
-        * ``remove_products``
-
-        """
-        for which in ('add_products', 'remove_products'):
-            path = entities.SyncPlan(
-                self.server_config,
-                id=2,
-                organization=1,
-            ).path(which)
-            self.assertIn(
-                'organizations/1/sync_plans/2/{0}'.format(which),
-                path
-            )
-            self.re_assertion(path, '{0}$'.format(which))
-
-    def test_system_path(self):
-        """Test :meth:`nailgun.entities.System.path`.
-
-        Assert that correct paths are returned when:
 
-        * A UUID is provided and ``which`` is omitted.
-        * A UUID is provided and ``which='self'``.
-        * A UUID is omitted and ``which`` is omitted.
-        * A UUID is omitted and ``which='base'``.
-
-        """
-        for gen_path in (
-                entities.System(self.server_config, uuid=self.id_).path(),
-                entities.System(self.server_config, uuid=self.id_).path(
-                    which='self'
-                )
-        ):
-            self.assertIn('/systems/{0}'.format(self.id_), gen_path)
-            self.re_assertion(gen_path, '{0}$'.format(self.id_))
-        for gen_path in (
-                entities.System(self.server_config).path(),
-                entities.System(self.server_config).path(which='base')):
-            self.assertIn('/systems', gen_path)
-            self.re_assertion(gen_path, 'systems$')
+# This file is divided in to three sets of test cases (`TestCase` subclasses):
+#
+# 1. Tests for inherited methods.
+# 2. Tests for entity-specific methods.
+# 3. Other tests.
+#
+# 1. Tests for inherited methods. ---------------------------------------- {{{1
```

Comment:
The curious `{{{1` marker is a manual code folding marker. Several text editors recognize it, such as vim, emacs and jEdit. What I've done here is added in some folding markers that divide the module in to logical sections instead of syntactical sections.

Under vim, try this:

    :set foldmethod=market
    [go to some section of the file]
    zc
    zo